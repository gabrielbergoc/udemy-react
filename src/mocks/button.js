export const buttonProps = {
  text: 'button text',
  title: 'button title',
  onClick: jest.fn(),
  disabled: false,
};
