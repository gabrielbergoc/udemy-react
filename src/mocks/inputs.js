export const searchInputProps = {
  value: "a value",
  handleChange: jest.fn(),
  placeholder: "a placeholder",
}
