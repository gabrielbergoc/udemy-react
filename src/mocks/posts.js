export const posts = [
  {
    id: 1,
    title: "post 1 title",
    body: "post 1 body",
    cover: "img/img1.png",
  },
  {
    id: 2,
    title: "post 2 title",
    body: "post 2 body",
    cover: "img/img2.png",
  },
  {
    id: 3,
    title: "post 3 title",
    body: "post 3 body",
    cover: "img/img3.png",
  },
];

export const post = posts[0];
