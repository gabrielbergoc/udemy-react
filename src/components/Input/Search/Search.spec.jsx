import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchInput from '.';
import { searchInputProps } from '../../../mocks/inputs';

describe('<SearchInput />', () => {
  it('should render correctly', () => {
    render(<SearchInput {...searchInputProps} />);

    const input = screen.getByPlaceholderText(searchInputProps.placeholder);

    expect(input).toHaveAttribute('type', 'search');
    expect(input).toHaveClass('search-input');
    expect(input).toHaveValue(searchInputProps.value);
  });

  it('should match snapshot', () => {
    const { container } = render(<SearchInput {...searchInputProps} />);

    expect(container).toMatchSnapshot();
  });

  it('should call handleChange on each keypress', () => {
    render(<SearchInput {...searchInputProps} />);
    const userInput = 'user input';
    const fn = searchInputProps.handleChange;
    const input = screen.getByPlaceholderText(searchInputProps.placeholder);

    userEvent.type(input, userInput);

    expect(fn).toHaveBeenCalledTimes(userInput.length);
  });
});
