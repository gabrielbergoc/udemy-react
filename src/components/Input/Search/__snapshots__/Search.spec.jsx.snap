// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match snapshot 1`] = `
<div>
  <input
    class="search-input"
    placeholder="a placeholder"
    type="search"
    value="a value"
  />
</div>
`;
